subprojects/core/src/main/java/org/gradle/internal/build/IncludedBuildFactory.java

```diff
@@ -22,4 +22,7 @@
 
 public interface IncludedBuildFactory {
     IncludedBuildState createBuild(BuildIdentifier buildIdentifier, Path identityPath, BuildDefinition buildDefinition, boolean isImplicit, BuildState owner);
+
+    default void prepareBuild(IncludedBuildState includedBuild) {
```

Comment:
Can you maybe add a comment why this method is need? (either here or in the CC specific subclass?)

Reply:
Yes, this is a good idea, done.